api/views/personal_info.py

```diff
@@ -14,8 +16,41 @@
 from api.serializers.errors import BaseAPIExceptionSerializer, ValidationErrorSerializer
 
 
+@extend_schema(
+    parameters=[
+        OpenApiParameter(
+            name="registration_telegram_bot_chat_id",
+            location=OpenApiParameter.QUERY,
+            type=int,
+            description="Filter by chat ID in Telegram registration bot",
+        )
+    ],
+    responses={
+        status.HTTP_200_OK: OpenApiResponse(
+            response=PersonalInfoSerializer, description="Returns personal info records."
+        ),
+        status.HTTP_406_NOT_ACCEPTABLE: OpenApiResponse(
+            response=BaseAPIExceptionSerializer,
+            description="No personal info records were found that match given params.",
+        ),
+    },
+)
 class PersonalInfoViewSet(viewsets.ModelViewSet[PersonalInfo]):
-    queryset = PersonalInfo.objects.all()
+    def get_queryset(self) -> QuerySet[PersonalInfo]:
+        """Optionally restricts the returned personal info items
+        to items with given chat ID in Telegram registration bot.
+        """
+        queryset: QuerySet[PersonalInfo] = PersonalInfo.objects.all()
+        chat_id = self.request.query_params.get("registration_telegram_bot_chat_id")
```

Comment:
Let's use django-filters please :) 

Like here:
https://github.com/samanthasgroup/django-webapps/blob/a0f9df55f749140883bc8836741daa6c5b67f3e8/api/filters/age_range.py#L6

Then you won't need to redefine get_queryset at all

Reply:
Done